src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -3374,8 +3374,20 @@ void CodeGen::genFloatToIntCast(GenTree* treeNode)
     }
 
     genConsumeOperands(treeNode->AsOp());
+    regNumber tmpReg = rsGetRsvdReg();
 
     GetEmitter()->emitIns_R_R(ins, EA_8BYTE, treeNode->GetRegNum(), op1->GetRegNum());
+
+    // This part emulates the "flush to zero" option because the RISC-V specification does not provide it.
+    instruction feq_ins = INS_feq_s;
+    if (srcType == TYP_DOUBLE)
+    {
+        feq_ins = INS_feq_d;
+    }
+    GetEmitter()->emitIns_R_R_R(feq_ins, EA_8BYTE, tmpReg, op1->GetRegNum(), op1->GetRegNum());
+    GetEmitter()->emitIns_R_R_R(INS_sub, EA_8BYTE, tmpReg, REG_R0, tmpReg);
+    GetEmitter()->emitIns_R_R_R(INS_and, EA_8BYTE, treeNode->GetRegNum(), treeNode->GetRegNum(), tmpReg);
```

Comment:
This likely needs a comment explaining what its doing.

`feq_s` compares `op1` to itself. I'm assuming this produces `0` for `false` and `1` for `true`. This means `0` for `NaN` and `1` for anything else.

For `sub`, i'm presuming that `REG_R0` (if so, an alias might really help readability/intent) is always zero and so we're doing `result = 0 - result`, thus producing `0` for `NaN` and `-1` (AllBitsSet) for `true`

`and` then produces `0` for `NaN` and preserves the original for anything else.

Reply:
Done.